File: src/Web/Admin/sakai-vue/src/service/TopicService.js
export default class TopicService {
    constructor(request) {
        this.request = request;
    }

    async getList(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/topic/list',
            params: filter
        });
        return response?.data ?? response;
    }

    async getById(id) {
        const response = await this.request({
            method: 'get',
            url: `/topic/${id}`
        });
        return response?.data ?? response;
    }

    async create(payload) {
        await this.request({
            method: 'post',
            url: '/topic/create',
            data: payload
        });
    }

    async update(payload) {
        await this.request({
            method: 'post',
            url: '/topic/update',
            data: payload
        });
    }

    async delete(topicId) {
        return this.request({ method: 'delete', url: `/topic/delete/${topicId}` });
    }

    async getLookup(filter = {}) {
        const response = await this.request({
            method: 'get',
            url: '/topic/lookup',
            params: filter
        });
        return response?.data ?? response;
    }
}

File: src/Mobile/lib/services/topic_service.dart
import 'package:dio/dio.dart';

/// WebApi `TopicService.js`
class TopicService {
  TopicService(this._dio);

  final Dio _dio;

  Future<dynamic> getList([Map<String, dynamic>? filter]) async {
    final r = await _dio.get<dynamic>('/topic/list', queryParameters: filter);
    return r.data;
  }

  Future<dynamic> getById(int id) async {
    final r = await _dio.get<dynamic>('/topic/$id');
    return r.data;
  }

  Future<dynamic> create(Map<String, dynamic> payload) async {
    final r = await _dio.post<dynamic>('/topic/create', data: payload);
    return r.data;
  }

  Future<dynamic> update(Map<String, dynamic> payload) async {
    final r = await _dio.post<dynamic>('/topic/update', data: payload);
    return r.data;
  }
}
